cli/main.go

```diff
@@ -160,35 +173,33 @@ func initConfig() {
 	flag.BoolVar(&setExitStatus, "set_exit_status", false, exitStatusUsage)
 	flag.IntVar(&maxOpenFiles, "max_open_files", 0, maxOpenFilesUsage)
 	flag.Parse()
+}
 
-	// Output build info (version, commit, date and builtBy)
-	if versionFlag {
-		var buildInfo string
-		if date != "unknown" && builtBy != "unknown" {
-			buildInfo = fmt.Sprintf("Built\t\t%s by %s\n", date, builtBy)
-		}
+// getVersion returns build info (version, commit, date and builtBy)
+func getVersion(builtBy, date, commit, version string) string {
+	var buildInfo string
+	if date != defaultDate && builtBy != defaultBuilder {
+		buildInfo = fmt.Sprintf("Built\t\t%s by %s\n", date, builtBy)
+	}
 
-		if commit != "none" {
-			buildInfo = fmt.Sprintf("Commit:\t\t%s\n%s", commit, buildInfo)
-		}
+	if commit != defaultCommit {
+		buildInfo = fmt.Sprintf("Commit:\t\t%s\n%s", commit, buildInfo)
+	}
 
-		if version == "dev" {
-			bi, ok := debug.ReadBuildInfo()
-			if ok {
-				version = bi.Main.Version
-				if strings.HasPrefix(version, "v") {
-					version = bi.Main.Version[1:]
-				}
-				if len(buildInfo) == 0 {
-					fmt.Printf("version %s\n", version)
-					os.Exit(0)
-				}
+	if version == defaultVersion {
+		bi, ok := debug.ReadBuildInfo()
+		if ok {
+			version = bi.Main.Version
+			if strings.HasPrefix(version, "v") {
+				version = strings.TrimLeft(bi.Main.Version, "v")
+			}
```

Comment:
You had no need to use HasPrefix + TrimLeft, my point was to use

```suggestion
			version = bi.Main.Version
			if trimedVersion, ok := strings.CutPrefix(version, "v"); ok {
				version = trimedVersion
			}
```

But because this would be complicated, it could be simply replaced by

```suggestion
			version := strings.TrimLeft(bi.Main.Version, "v")
```

Here, no `if` is involved

Reply:
Created #1137